test/argentWalletDetector.js

```diff
@@ -0,0 +1,81 @@
+    proxy2 = await deployer.deploy(Proxy, {}, implementation2.contractAddress);
+  });
+
+  beforeEach(async () => {
+    detector = await deployer.deploy(ArgentWalletDetector, {}, [], []);
+  });
+
+  describe("add info", () => {
+    it("should deploy with codes and implementations", async () => {
+      const c = [argentCode, randomCode];
+      const i = [implementation1.contractAddress, implementation2.contractAddress];
+      detector = await deployer.deploy(ArgentWalletDetector, {}, c, i);
+      const implementations = await detector.getImplementations();
+      assert.equal(implementations[0], implementation1.contractAddress, "should have added first implementation");
+      assert.equal(implementations[1], implementation2.contractAddress, "should have added second implementation");
+      const codes = await detector.getCodes();
+      assert.equal(codes[0], argentCode, "should have added first code");
+      assert.equal(codes[1], randomCode, "should have added second code");
+    });
+
+    it("should add implementations", async () => {
+      await detector.addImplementation(implementation1.contractAddress);
+      await detector.addImplementation(implementation2.contractAddress);
+      const implementations = await detector.getImplementations();
+      assert.equal(implementations[0], implementation1.contractAddress, "should have added first implementation");
+      assert.equal(implementations[1], implementation2.contractAddress, "should have added second implementation");
+    });
+
+    it("should add codes", async () => {
+      await detector.addCode(argentCode);
+      await detector.addCode(randomCode);
+      const codes = await detector.getCodes();
+      assert.equal(codes[0], argentCode, "should have added first code");
+      assert.equal(codes[1], randomCode, "should have added second code");
+    });
+
+    it("should add code and implementation from a wallet", async () => {
+      await detector.addCodeAndImplementationFromWallet(proxy1.contractAddress);
+      const isArgent = await detector.isArgentWallet(proxy1.contractAddress);
+      assert.isTrue(isArgent, "should return true for an Argent wallet");
+    });
+
+    it("should return false when the code is not correct", async () => {
+      await detector.addImplementation(implementation1.contractAddress);
+      await detector.addCode(argentCode);
+      const isArgent = await detector.isArgentWallet(detector.contractAddress);
+      assert.isFalse(isArgent, "should return false when the code is not correct");
+    });
+
+    it("should return false when the implementation is not correct", async () => {
+      await detector.addImplementation(implementation1.contractAddress);
+      await detector.addCode(argentCode);
+      const isArgent = await detector.isArgentWallet(proxy2.contractAddress);
+      assert.isFalse(isArgent, "should return false when the implementation is not correct");
```

Comment:
This test is not succeeding for the right reason. The code hash of proxy2 is actually `0xb150b4da11d3a51d25092296fb52138fc2d22faa4191d25d4d2e99043f9a28b1`, which is not what is stored in `argentCode`. As a result, the test succeeds because `acceptedCodes[codeHash].exists` is `false`, not because `acceptedImplementations[IWallet(_wallet).implementation()].exists` is `false`. I'd suggest replacing
```
await detector.addCode(argentCode);
```
with:
```
await detector.addCode(ethers.utils.keccak256(proxy2._contract.deployedBytecode));
```
which will make sure we are adding the correct code hash for `proxy2`.